extensions/sql-migration/src/api/sqlUtils.ts

```diff
@@ -509,3 +510,73 @@ export async function isSourceConnectionSysAdmin(): Promise<boolean> {
+	interface StorageAccountAdditionalProperties {
+		publicNetworkAccess: string,
+		networkAcls: NetworkRuleSet,
+		privateEndpointConnections: PrivateEndpointConnection[]
+	}
+	interface NetworkRuleSet {
+		virtualNetworkRules: VirtualNetworkRule[],
+		defaultAction: string
+	}
+	interface VirtualNetworkRule {
+		id: string,
+		state: string,
+		action: string
+	}
+
+	const storageAccountProperties: StorageAccountAdditionalProperties = (storageAccount as any)['properties'];
+	const storageAccountPublicAccessEnabled: boolean = storageAccountProperties.publicNetworkAccess ? storageAccountProperties.publicNetworkAccess.toLowerCase() === 'Enabled'.toLowerCase() : true;
+	const storageAccountDefaultIsAllow: boolean = storageAccountProperties.networkAcls ? storageAccountProperties.networkAcls.defaultAction.toLowerCase() === 'Allow'.toLowerCase() : true;
+	const storageAccountWhitelistedVNets: string[] = storageAccountProperties.networkAcls ? storageAccountProperties.networkAcls.virtualNetworkRules.filter(rule => rule.action.toLowerCase() === 'Allow'.toLowerCase()).map(rule => rule.id) : [];
+
+	var enabledFromAllNetworks: boolean = false;
+	var enabledFromWhitelistedVNet: boolean = false;
+	var enabledFromPrivateEndpoint: boolean = false;
+
+	// 1) check for access from all networks
+	enabledFromAllNetworks = storageAccountPublicAccessEnabled && storageAccountDefaultIsAllow;
+
+	switch (targetType) {
+		case MigrationTargetType.SQLMI:
+			const targetManagedInstanceVNet: string = (targetServer.properties as any)['subnetId'] ?? '';
+			const targetManagedInstancePrivateEndpointConnections: PrivateEndpointConnection[] = (targetServer.properties as any)['privateEndpointConnections'] ?? [];
+			const storageAccountPrivateEndpointConnections: PrivateEndpointConnection[] = storageAccountProperties.privateEndpointConnections ?? [];
+
+			// 2) check for access from whitelisted vnet
+			if (storageAccountWhitelistedVNets.length > 0) {
+				enabledFromWhitelistedVNet = storageAccountWhitelistedVNets.some(vnet => vnet.toLowerCase() === targetManagedInstanceVNet.toLowerCase());
+			}
+
+			// 3) check for access from private endpoint
+			if (targetManagedInstancePrivateEndpointConnections.length > 0) {
+				enabledFromPrivateEndpoint = storageAccountPrivateEndpointConnections.some(async privateEndpointConnection => {
+					const privateEndpoint = await NetworkInterfaceModel.getPrivateEndpoint(account, subscription, privateEndpointConnection.id);
+					const privateEndpointSubnet = privateEndpoint.properties.subnet ? privateEndpoint.properties.subnet.id : '';
+					return NetworkInterfaceModel.getVirtualNetworkFromSubnet(privateEndpointSubnet).toLowerCase() === NetworkInterfaceModel.getVirtualNetworkFromSubnet(targetManagedInstanceVNet).toLowerCase();
+				});
+			}
+
+			break;
+		case MigrationTargetType.SQLVM:
+			// to=do: VM scenario -- for VM, get subnet by first checking underlying compute VM, then its network interface
```

Comment:
can these comments be removed? or will it be used in this PR?

Reply:
was going to address VM scenario in a separate PR since VM occurrence of this error is much lower, but implementation should basically be uncommenting this part plus a few additional checks and it should be good. But since it hasn't been tested yet, will remove this comment for now

Reply:
removed the unnecessary comments but left the todo